### DISCUSSION TITLE
Sveltekit monorepo server-side only packages

### DISCUSSION THREAD

**Discussion by kevbook** [sveltejs/kit#10899] [General]
I have a sveltekit monorepo with the following structure. There is a server-side package (`/packages/server-package1`). How can I ensure that this module behaves as if it were located in the `website1/src/lib/server` directory? (ie. it can't be imported on components or client-side code). 

Note: Using pnpm workspace to build this monorepo structure.

```
| [root]
|── apps
    |── website1 (SvelteKit app)
    |── website2 (SvelteKit app)
|── packages
    |── server-package1
```

**Comment by steve-taylor** [comment]
Assuming you're using TypeScript, add these rules and overrides to `apps/website1/.eslintrc`:

```json
{
  "rules": {
    "@typescript-eslint/no-restricted-imports": [
      "error",
      {
        "paths": [
          {"name": "server-package1"}
        ]
      }
    ]
  },
  "overrides": [
    {
      "files": "src/lib/server/**/*.ts",
      "rules": {
        "@typescript-eslint/no-restricted-imports": ["off"]
      }
    }
  ]
}
```

Re-export everything in `server-package1`:
```typescript
// src/lib/server/server-package1.ts

export * from 'server-package1'
```

Then you can use it like so:
```typescript
// Replaces import {foo, bar} from 'server-package1'
import {foo, bar} from '$lib/server/server-package1'
```
